fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -2913,6 +2919,20 @@ public void createMaterializedView(CreateMaterializedViewStatement stmt)
         } catch (AnalysisException e) {
             throw new DdlException(e.getMessage());
         }
+        // validate optHints
+        Map<String, String> optHints = null;
+        QueryRelation queryRelation = stmt.getQueryStatement().getQueryRelation();
+        if (queryRelation instanceof SelectRelation) {
+            SelectRelation selectRelation = (SelectRelation) queryRelation;
+            optHints = selectRelation.getSelectList().getOptHints();
+            if (optHints != null && !optHints.isEmpty()) {
+                SessionVariable sessionVariable = VariableMgr.newSessionVariable();
```

Comment:
could you get the ConnectContext here? if could, we could get the SessionVariable from the ConnectContext.

Reply:
optHints here are passed to task, not used by current ConnectContext. And this SessionVariable is used to checked the validity of optHints, so I use a new temp SessionVariable here.